core/src/main/java/org/apache/accumulo/core/fate/accumulo/AccumuloStore.java

```diff
@@ -268,7 +269,9 @@ public void pop() {
       verifyReserved(true);
 
       Optional<Integer> top = findTop();
-      top.ifPresent(t -> newMutator(fateId).deleteRepo(t).mutate());
+      top.ifPresent(t -> newMutator(fateId)
+          .requireStatus(TStatus.FAILED_IN_PROGRESS, TStatus.SUCCESSFUL, TStatus.FAILED)
```

Comment:
Ideally we would only pop when something has a status of FAILED_IN_PROGRESS.  Do you know if this change would cause test failures?

```suggestion
          .requireStatus(TStatus.FAILED_IN_PROGRESS)
```

Reply:
It would cause test failures but we can just set the status of the txStore in the test before this operation.

Reply:
Fixed in 2c2c161